sklearn/linear_model/cd_fast.pyx

```diff
@@ -105,29 +106,44 @@ cdef extern from "cblas.h":
 
     void daxpy "cblas_daxpy"(int N, double alpha, double *X, int incX,
                              double *Y, int incY) nogil
+    void saxpy "cblas_saxpy"(int N, float alpha, float *X, int incX,
+                             float *Y, int incY) nogil
     double ddot "cblas_ddot"(int N, double *X, int incX, double *Y, int incY
                              ) nogil
+    float sdot "cblas_sdot"(int N, float *X, int incX, float *Y, int incY
+                             ) nogil
     double dasum "cblas_dasum"(int N, double *X, int incX) nogil
+    float sasum "cblas_sasum"(int N, float *X, int incX) nogil
     void dger "cblas_dger"(CBLAS_ORDER Order, int M, int N, double alpha,
                 double *X, int incX, double *Y, int incY, double *A, int lda) nogil
+    void sger "cblas_sger"(CBLAS_ORDER Order, int M, int N, float alpha,
+                float *X, int incX, float *Y, int incY, float *A, int lda) nogil
     void dgemv "cblas_dgemv"(CBLAS_ORDER Order,
                       CBLAS_TRANSPOSE TransA, int M, int N,
                       double alpha, double *A, int lda,
                       double *X, int incX, double beta,
                       double *Y, int incY) nogil
+    void sgemv "cblas_sgemv"(CBLAS_ORDER Order,
+                      CBLAS_TRANSPOSE TransA, int M, int N,
+                      float alpha, float *A, int lda,
+                      float *X, int incX, float beta,
+                      float *Y, int incY) nogil
     double dnrm2 "cblas_dnrm2"(int N, double *X, int incX) nogil
+    float snrm2 "cblas_snrm2"(int N, float *X, int incX) nogil
     void dcopy "cblas_dcopy"(int N, double *X, int incX, double *Y, int incY) nogil
+    void scopy "cblas_scopy"(int N, float *X, int incX, float *Y, int incY) nogil
     void dscal "cblas_dscal"(int N, double alpha, double *X, int incX) nogil
+    void sscal "cblas_sscal"(int N, float alpha, float *X, int incX) nogil
 
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
 @cython.cdivision(True)
-def enet_coordinate_descent(np.ndarray[DOUBLE, ndim=1] w,
-                            double alpha, double beta,
-                            np.ndarray[DOUBLE, ndim=2, mode='fortran'] X,
-                            np.ndarray[DOUBLE, ndim=1, mode='c'] y,
-                            int max_iter, double tol,
+def enet_coordinate_descent(np.ndarray[floating, ndim=1] w,
+                            floating alpha, floating beta,
+                            np.ndarray[floating, ndim=2, mode='fortran'] X,
+                            np.ndarray[floating, ndim=1, mode='c'] y,
```

Comment:
I'm not sure that we need to support `y`, `alpha`, `beta`, `w` or `tol` with lower precision than `DOUBLE`. Or is consistent type necessary to use BLAS?